src/devdocs2zim/generator.py

```diff
@@ -491,6 +516,39 @@ def generate_zim(
             )
         return zim_path
 
+    @staticmethod
+    def fetch_logo_bytes(user_logo_path: str) -> bytes:
+        """Fetch a user-supplied logo for the ZIM and format/resize it.
+
+        Parameters:
+          user_logo_path: Path or URL to the logo.
+        """
+        logger.info(f"  Fetching logo from: {user_logo_path}")
+        full_logo_path: Path | None = handle_user_provided_file(source=user_logo_path)
+        if full_logo_path is None:
+            logger.warning("  Fetching logo failed, using fallback.")
+            full_logo_path = DEFAULT_LOGO_PATH
+
+        # convert to PNG
+        png_logo_path = Path(
+            tempfile.NamedTemporaryFile(suffix=".png", delete=False).name
+        )
+        try:
+            convert_image(full_logo_path, png_logo_path, fmt="PNG")
+        except UnidentifiedImageError:
```

Comment:
I don't like this much, the image is not necessarily an SVG when it is not identified. Did you saw we have the `format_for` method available in zimscraperlib, and it supports proper detection of SVGs? (you probably need to pass `from_suffix = False` since suffix might be misleading in this use case.

Reply:
fixed